File: src/main/java/team/zeds/ancientmagic/item/ManaStorage.java
package team.zeds.ancientmagic.item;

import team.zeds.ancientmagic.api.MagicItem;
import team.zeds.ancientmagic.api.magic.MagicTypes;

public class ManaStorage extends MagicItem {
    public ManaStorage(MagicBuilder builder, int manaCount, boolean fireProof) {
        super(builder.setMagicSubtype(MagicTypes.STORAGE).setMaxMana(manaCount));

        if (fireProof) builder.fireProof();
    }
}

File: src/main/java/team/zeds/ancientmagic/item/ManaStorage.kt
package team.zeds.ancientmagic.item

import team.zeds.ancientmagic.api.item.MagicItem
import team.zeds.ancientmagic.api.item.MagicItemBuilder
import team.zeds.ancientmagic.api.magic.MagicTypes

class ManaStorage(builder: MagicItemBuilder, manaCount: Int, fireProof: Boolean) :
    MagicItem(builder.setMagicSubtype(MagicTypes.STORAGE).setMaxMana(manaCount)) {
    init {
        if (fireProof) builder.fireProof()
    }
}
